tests/integration/helpers.py

```diff
@@ -0,0 +1,289 @@
+import datetime
+import logging
+import re
+import subprocess
+import uuid
+from pathlib import Path
+
+import yaml
+from pytest_operator.plugin import OpsTest
+
+from constants import (
+    HA_ZNODE_NAME,
+)
+
+logger = logging.getLogger(__name__)
+
+METADATA = yaml.safe_load(Path("./metadata.yaml").read_text())
+APP_NAME = METADATA["name"]
+ZOOKEEPER_NAME = "zookeeper-k8s"
+TEST_CHARM_PATH = "./tests/integration/app-charm"
+TEST_CHARM_NAME = "application"
+
+PROCESS_NAME_PATTERN = "org.apache.kyuubi.server.KyuubiServer"
+KYUUBI_CONTAINER_NAME = "kyuubi"
+
+
+def get_random_name():
+    return str(uuid.uuid4()).replace("-", "_")
+
+
+async def fetch_jdbc_endpoint(ops_test):
+    logger.info("Running action 'get-jdbc-endpoint' on kyuubi-k8s unit...")
+    kyuubi_unit = ops_test.model.applications[APP_NAME].units[0]
+    action = await kyuubi_unit.run_action(
+        action_name="get-jdbc-endpoint",
+    )
+    result = await action.wait()
+
+    jdbc_endpoint = result.results.get("endpoint")
+    logger.info(f"JDBC endpoint: {jdbc_endpoint}")
+
+    return jdbc_endpoint
+
+
+async def run_sql_test_against_jdbc_endpoint(ops_test: OpsTest, test_pod):
+    """Verify the JDBC endpoint exposed by the charm with some SQL queries."""
+    jdbc_endpoint = await fetch_jdbc_endpoint(ops_test)
+    database_name = get_random_name()
+    table_name = get_random_name()
+    logger.info(
+        "Testing JDBC endpoint by connecting with beeline and executing a few SQL queries. "
+        f"Using database {database_name} and table {table_name} ..."
+    )
+    process = subprocess.run(
+        [
+            "./tests/integration/test_jdbc_endpoint.sh",
+            test_pod,
+            ops_test.model_name,
+            jdbc_endpoint,
+            get_random_name(),
+            get_random_name(),
+        ],
+        capture_output=True,
+    )
+    print("========== test_jdbc_endpoint.sh STDOUT =================")
+    print(process.stdout.decode())
+    print("========== test_jdbc_endpoint.sh STDERR =================")
+    print(process.stderr.decode())
+    logger.info(f"JDBC endpoint test returned with status {process.returncode}")
+    return process.returncode == 0
+
+
+# async def get_kyuubi_znode_children(ops_test: OpsTest, test_pod) -> list[str]:
```

Comment:
Do we need to keep this commented function?

Reply:
We don't. However, I kept this here because I wanted both of your and @deusebio's view on whether we should use the uncommented (`kyuubi-ctl` way) of fetching this information or this commented `zookeeper` way. To me, `ls znode` approach using zookeeper is way cleaner and straightforward, but I equally understand your preference over using `kyuubi` native tools to achieve the same result. @deusebio, what do you think?

Reply:
Uhm, I don't have strong opinions. Using the `kyuubi-ctl` is interesting since we would ALSO test that. But if you prefer to just use `kazoo` since this function is kyuubi independent, this makes sense too. 

Up to you really. But I would not use commented code, why not using a flag in the function `backend` and we provide BOTH implementation out of the box?